src/instructlab/sdg/generate_data.py

```diff
@@ -192,9 +200,21 @@ def load_pipeline(yaml_basename):
             return Pipeline.from_file(ctx, os.path.join(pipeline, yaml_basename))
 
     return (
-        SDG([load_pipeline("knowledge.yaml")]),
-        SDG([load_pipeline("freeform_skills.yaml")]),
-        SDG([load_pipeline("grounded_skills.yaml")]),
+        SDG(
+            [load_pipeline("knowledge.yaml")],
+            num_workers=num_workers,
+            batch_size=batch_size,
```

Comment:
Right, that makes a lot of sense if we push this logic down to `Pipeline`. If we do go that route, I wonder about making `PipelineContext` more feature-full and having it own the lifecycle of a single global `ThreadPoolExecutor`. This could allow the individual `Blocks` to submit their own futures while maintaining centralized control over the number of concurrent threads. If we opted to use a `ProcessPoolExecutor` in the future, it would also be a single place make the swap.

Reply:
I'm fine with not getting too fancy for now - we're only going to be using one pipeline at a time anyway